pkg/runtime/reconciler.go

```diff
@@ -629,41 +621,45 @@ func getPatchDocument(
 // patchResourceMetadataAndSpec patches the custom resource in the Kubernetes
 // API to match the supplied latest resource's metadata and spec.
 //
-// NOTE(jaypipes): The latest parameter is *mutated* by this method: the
-// resource's metadata.resourceVersion is incremented in the process of calling
-// Patch. This is intentional, because without updating the resource's
-// metadata.resourceVersion, the resource cannot be passed to Patch again later
-// in the reconciliation loop if Patch is called with the Optimistic Locking
-// option.
+// NOTE(redbackthomson): This method returns an updated version of the latest
+// parameter. This version has an updated metadata.resourceVersion, which is
+// incremented in the process of calling Patch. This is intentional, because
+// without updating the resource's metadata.resourceVersion, the resource cannot
+// be passed to Patch again later in the reconciliation if Patch is called with
+// the Optimistic Locking option.
```

Comment:
Perhaps update this note to say that the returned resource has its resource references removed?